File: src/main/java/com/atsuishio/superbwarfare/init/ModCriteriaTriggers.java
package com.atsuishio.superbwarfare.init;

import com.atsuishio.superbwarfare.Mod;
import com.atsuishio.superbwarfare.advancement.criteria.OttoSprintTrigger;
import com.atsuishio.superbwarfare.advancement.criteria.RPGMeleeExplosionTrigger;
import net.minecraft.advancements.CriterionTrigger;
import net.minecraft.core.Registry;
import net.minecraft.core.registries.BuiltInRegistries;

public class ModCriteriaTriggers {

    public static final RPGMeleeExplosionTrigger RPG_MELEE_EXPLOSION = register("rpg_melee_explosion", new RPGMeleeExplosionTrigger());
    public static final OttoSprintTrigger OTTO_SPRINT = register("otto_sprint", new OttoSprintTrigger());

    private static <T extends CriterionTrigger<?>> T register(String name, T trigger) {
        return Registry.register(BuiltInRegistries.TRIGGER_TYPES, Mod.loc(name), trigger);
    }

    public static void init() {
    }
}

File: src/main/kotlin/com/atsuishio/superbwarfare/init/ModCriteriaTriggers.kt
package com.atsuishio.superbwarfare.init

import com.atsuishio.superbwarfare.Mod
import com.atsuishio.superbwarfare.advancement.criteria.OttoSprintTrigger
import com.atsuishio.superbwarfare.advancement.criteria.RPGMeleeExplosionTrigger
import com.atsuishio.superbwarfare.advancement.criteria.VehicleHurtTrigger
import net.minecraft.advancements.CriterionTrigger
import net.minecraft.core.Registry
import net.minecraft.core.registries.BuiltInRegistries

object ModCriteriaTriggers {
    @JvmField
    val RPG_MELEE_EXPLOSION: RPGMeleeExplosionTrigger = register("rpg_melee_explosion", RPGMeleeExplosionTrigger())!!

    @JvmField
    val OTTO_SPRINT: OttoSprintTrigger = register("otto_sprint", OttoSprintTrigger())!!

    @JvmField
    val VEHICLE_HURT: VehicleHurtTrigger = register("vehicle_hurt", VehicleHurtTrigger())!!

    private fun <T : CriterionTrigger<*>?> register(name: String?, trigger: T?): T? {
        return Registry.register<CriterionTrigger<*>?, T?>(BuiltInRegistries.TRIGGER_TYPES, Mod.loc(name!!), trigger)
    }

    @JvmStatic
    fun init() {
    }
}
